src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -763,13 +763,18 @@ private uint getMethodAttribsInternal(MethodDesc method)
                 result |= CorInfoFlag.CORINFO_FLG_FORCEINLINE;
             }
 
-            if (method.OwningType.IsDelegate)
-            {
-                if (method.Name == "Invoke")
-                    // This is now used to emit efficient invoke code for any delegate invoke,
-                    // including multicast.
-                    result |= CorInfoFlag.CORINFO_FLG_DELEGATE_INVOKE;
-            }
+            if (method.OwningType.IsDelegate && method.Name == "Invoke")
+            {
+                // This is now used to emit efficient invoke code for any delegate invoke,
+                // including multicast.
+                result |= CorInfoFlag.CORINFO_FLG_DELEGATE_INVOKE;
+
+                // Make sure to report the method as final if it's final in the metadata.
+                // RyuJIT special cases this method; it would assert if it's not final
+                // and we might not have set the bit in the code above.
+                if (method.IsFinal || method.OwningType.IsSealed())
```

Comment:
Do we need this condition at all? Why not set the flag unconditionally?

Reply:
It felt more correct to report what's in the metadata as opposed to sanitizing user input in an attempt to make it work. Granted, not reporting just leads to an assert, and RyuJIT won't treat it as bad code so maybe it doesn't help much...